src/sentry/consumers/__init__.py

```diff
@@ -394,12 +396,19 @@ def get_stream_processor(
         ) from e
 
     strategy_factory_cls = import_string(consumer_definition["strategy_factory"])
-    logical_topic = consumer_definition["topic"]
-    if not isinstance(logical_topic, str):
-        logical_topic = logical_topic()
+    consumer_topic = consumer_definition["topic"]
+    if isinstance(consumer_topic, Topic):
+        default_topic = consumer_topic.value
+        real_topic = settings.KAFKA_TOPIC_OVERRIDES.get(default_topic, default_topic)
+    else:
+        # TODO: Deprecated, remove once this way is no longer used
+        if not isinstance(consumer_topic, str):
```

Comment:
This condition seems counterintuitive. Was that any reason not to phrase it positively as:
```
if isinstance(consumer_topic, str):
    real_topic = consumer_topic
else:
   ....
```

Reply:
I did not write this code, I copied it from line 398 above, and don't want to refactor it unnecessarily given it will probably go away in a couple of days.